kale/pipeline/multiomics_trainer.py

```diff
@@ -0,0 +1,232 @@
+# =============================================================================
+# Author: Sina Tabakhi, sina.tabakhi@gmail.com
+# =============================================================================
+
+"""
+Construct a pipeline to run the MOGONET method based on PyTorch Lightning.
+
+This code is written by refactoring the MOGONET code (https://github.com/txWang/MOGONET/blob/main/train_test.py)
+within the PyTorch Lightning.
+
+Reference:
+Wang, T., Shao, W., Huang, Z., Tang, H., Zhang, J., Ding, Z., Huang, K. (2021). MOGONET integrates multi-omics data
+using graph convolutional networks allowing patient classification and biomarker identification. Nature communications.
+https://www.nature.com/articles/s41467-021-23774-w
+"""
+
+from typing import List, Optional, Union
+
+import pytorch_lightning as pl
+import torch
+import torch.nn.functional as F
+from sklearn.metrics import accuracy_score, f1_score, roc_auc_score
+from torch import Tensor
+from torch.nn import CrossEntropyLoss, ModuleList
+from torch.optim.optimizer import Optimizer
+from torch_geometric.loader import DataLoader
+from torch_sparse import SparseTensor
+
+from kale.embed.mogonet import MogonetGCN
+from kale.loaddata.multiomics_datasets import SparseMultiOmicsDataset
+from kale.predict.decode import LinearClassifier, VCDN
+
+
+class ModalityTrainer(pl.LightningModule):
+    r"""The implementation of the MOGONET method, which is based on PyTorch Lightning.
```

Comment:
The sentence is not meaning what it is intended to. Discuss in person.

Reply:
Also need to explain what is this for. The implementation of MOGONET is already in embed with an API called mogonet.

Reply:
Changed.